hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanActionExecutor.java

```diff
@@ -147,7 +148,9 @@ List<HoodieCleanStat> clean(HoodieEngineContext context, HoodieCleanerPlan clean
     List<String> partitionsToBeDeleted = cleanerPlan.getPartitionsToBeDeleted() != null ? cleanerPlan.getPartitionsToBeDeleted() : new ArrayList<>();
     partitionsToBeDeleted.forEach(entry -> {
       try {
-        deleteFileAndGetResult(table.getMetaClient().getFs(), table.getMetaClient().getBasePath() + "/" + entry);
+        if (!StringUtils.isNullOrEmpty(entry)) {
+          deleteFileAndGetResult(table.getMetaClient().getFs(), table.getMetaClient().getBasePath() + "/" + entry);
```

Comment:
Kind of think the `cleanerPlan.getPartitionsToBeDeleted()` should be fixed, can we write a test case for it.

Reply:
Ideally, the `cleanerPlan.getPartitionsToBeDeleted()` should be empty for non-partitioned table and I added a test to verify whether the whole table is deleted or not. But, maybe, in an unlikely scenario, when table is partitioned but with empty partition path, this could happen.

Reply:
> in an unlikely scenario, when table is partitioned but with empty partition path, this could happen.

It actually happens inside my company🥲, and all data are deleted, but have no clue how it happened...

Reply:
Hi @leosanqing 

May I ask if you have encountered this issue? A writer generated data files incorrectly in table root path, but the table is a partitioned table.

E.g.

```
hdfs://..../hudi_table_folder/partition1/...parquet
hdfs://..../hudi_table_folder/partition2/...parquet
hdfs://..../hudi_table_folder/...parquet <- this file should not be here.
```

Reply:
> Hi @leosanqing
> 
> May I ask if you have encountered this issue? A writer generated data files incorrectly in table root path, but the table is a partitioned table.
> 
> E.g.
> 
> ```
> hdfs://..../hudi_table_folder/partition1/...parquet
> hdfs://..../hudi_table_folder/partition2/...parquet
> hdfs://..../hudi_table_folder/...parquet <- this file should not be here.
> ```

No, I haven't.

Reply:
Got it. Thanks for your reply.